src/sas/qtgui/Perspectives/Fitting/FittingWidget.py

```diff
@@ -57,6 +57,21 @@
 
 DEFAULT_POLYDISP_FUNCTION = 'gaussian'
 
+# CRUFT: remove when new release of sasmodels is available
+# https://github.com/SasView/sasview/pull/181#discussion_r218135162
+from sasmodels.sasview_model import SasviewModel
+if not hasattr(SasviewModel, 'get_weights'):
+    def get_weights(self, name):
+        """Returns model weights by parameter name.
+        type: (str) -> Tuple(float, np.ndarray, np.ndarray)
+        Supposed to be a member of SasviewModel."""
+        p = [p for p in self._model_info.parameters.call_parameters
+                        if name is p.name]
+        if not len(p):
+            return None, None, None
+        return self._get_weights(p[0])
```

Comment:
The parameter table has a `__getitem__()` method, so the following should suffice:
```
def get_weights(self, name):
    """
    Returns the polydispersity distribution for parameter *name* as *value* and *weight* arrays.
    """
    # type: (str) -> Tuple(np.ndarray, np.ndarray)
    _, x, w = self._get_weights(self._model_info.parameters[name])
    return x, w
```
I also simplified the interface so it doesn't return the centre, only the x, w arrays.